planning/behavior_path_planner/include/behavior_path_planner/scene_module/pull_over/shift_pull_over.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BEHAVIOR_PATH_PLANNER__SCENE_MODULE__PULL_OVER__SHIFT_PULL_OVER_HPP_
+#define BEHAVIOR_PATH_PLANNER__SCENE_MODULE__PULL_OVER__SHIFT_PULL_OVER_HPP_
+
+#include "behavior_path_planner/scene_module/pull_over/pull_over_path.hpp"
+#include "behavior_path_planner/scene_module/pull_over/pull_over_planner_base.hpp"
+#include "behavior_path_planner/scene_module/utils/occupancy_grid_based_collision_detector.hpp"
+
+#include <lane_departure_checker/lane_departure_checker.hpp>
+
+#include <autoware_auto_planning_msgs/msg/path_with_lane_id.hpp>
+
+#include <memory>
+#include <vector>
+
+namespace behavior_path_planner
+{
+using lane_departure_checker::LaneDepartureChecker;
+
+class ShiftPullOver : public PullOverPlannerBase
+{
+public:
+  explicit ShiftPullOver(
```

Comment:
same here for explicit

Reply:
Also shared_ptr with reference without constant should be fixed.

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/1911/commits/b47d458c9870cfcb74d01d9866159bd273ee1972